src/pages/Login.js

```diff
@@ -0,0 +1,149 @@
+import axios from "axios";
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useEffect } from "react";
+import "../components/css/login.css";
+import { notificationHandler } from "../components/Notification";
+import { useUserContext } from "../context & data/UserContext";
+
+const LoginPage = () => {
+  const [action, setAction] = useState("login");
+  const navigate = useNavigate();
+  const { dispatch } = useUserContext();
+
+  useEffect(() => {
+    if (
+      localStorage.getItem("isAuth") &&
+      localStorage.getItem("isAuth") !== "false" &&
+      localStorage.getItem("token").length > 25
+    )
+      navigate(-1);
+  }, [navigate]);
+
+  const submitHandler = (e) => {
+    e.preventDefault();
+    const formData = new FormData(e.target);
+    let data = {};
+    for (let x of formData) {
+      data[x[0]] = x[1];
+    }
+    axios
+      .post("/api/auth/login", data)
+      .then((res) => {
+        localStorage.setItem("isAuth", true);
+        localStorage.setItem("token", res.data.encodedToken);
+        notificationHandler({
+          type: "success",
+          content: "Successfully Logged In",
+        });
+        setTimeout(() => {
+          dispatch({ for: "userData", type: "update" });
+          navigate("/");
+        }, 2000);
+      })
+      .catch((err) => {
+        notificationHandler({
+          type: "warning",
+          content: "Login/ Password Incorrect",
+        });
+        console.error(err);
+      });
+  };
+
+  return (
+    <div className="section loginPage">
+      <div className="container">
+        {action === "login" ? (
+          <div className="login_comp flex flex-col p2">
+            <h2 className="head2">Login</h2>
+            <form className="login_form flex flex-col" onSubmit={submitHandler}>
+              <label htmlFor="username">Uername</label>
+              <div className="input">
+                <input type="text" id="username" name="email" required />
+              </div>
+              <label htmlFor="password">Password</label>
+              <div className="input">
+                <input type="password" id="password" name="password" required />
+              </div>
+              <label className="checkbox">
+                <input type="checkbox" />
+                Remember Me
+              </label>
+              <button className="btn-sec" type="submit">
+                Login
+              </button>
+              <button type="button" onClick={() => setAction("signup")}>
+                Create An Account {">"}
+              </button>
+            </form>
+          </div>
+        ) : (
+          <SignUp setAction={() => setAction("login")} />
+        )}
+      </div>
+    </div>
+  );
+};
+
+const SignUp = ({ setAction }) => {
```

Comment:
Is it possible to separate the util functions and components into separate files? If yes, you could try doing it. That would make your code cleaner.